packages/mds-utils/state-machine.ts

```diff
@@ -1,57 +1,147 @@
+    [VEHICLE_EVENTS.trip_cancel]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.trip_end]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.trip_leave_jurisdiction]: [VEHICLE_STATES.elsewhere],
+    [VEHICLE_EVENTS.missing]: [VEHICLE_STATES.unknown]
   },
   [VEHICLE_STATES.removed]: {
-    [VEHICLE_EVENTS.trip_enter]: VEHICLE_STATES.trip,
-    [VEHICLE_EVENTS.provider_drop_off]: VEHICLE_STATES.available,
-    [VEHICLE_EVENTS.deregister]: VEHICLE_STATES.inactive
+    [VEHICLE_EVENTS.agency_drop_off]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.decommissioned]: [VEHICLE_STATES.removed],
+    [VEHICLE_EVENTS.provider_drop_off]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.unspecified]: [VEHICLE_STATES.available]
   },
   [VEHICLE_STATES.reserved]: {
-    [VEHICLE_EVENTS.trip_start]: VEHICLE_STATES.trip,
-    [VEHICLE_EVENTS.cancel_reservation]: VEHICLE_STATES.available
-  },
-  [VEHICLE_STATES.trip]: {
-    [VEHICLE_EVENTS.trip_leave]: VEHICLE_STATES.elsewhere,
-    [VEHICLE_EVENTS.trip_end]: VEHICLE_STATES.available
+    [VEHICLE_EVENTS.comms_lost]: [VEHICLE_STATES.unknown],
+    [VEHICLE_EVENTS.missing]: [VEHICLE_STATES.unknown],
+    [VEHICLE_EVENTS.reservation_cancel]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.trip_start]: [VEHICLE_STATES.on_trip],
+    [VEHICLE_EVENTS.unspecified]: [VEHICLE_STATES.available]
   },
-  [VEHICLE_STATES.unavailable]: {
-    [VEHICLE_EVENTS.service_start]: VEHICLE_STATES.available,
-    [VEHICLE_EVENTS.deregister]: VEHICLE_STATES.inactive,
-    [VEHICLE_EVENTS.agency_pick_up]: VEHICLE_STATES.removed,
-    [VEHICLE_EVENTS.provider_pick_up]: VEHICLE_STATES.removed
+  [VEHICLE_STATES.unknown]: {
+    [VEHICLE_EVENTS.agency_drop_off]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.agency_pick_up]: [VEHICLE_STATES.removed],
+    [VEHICLE_EVENTS.comms_restored]: [
+      VEHICLE_STATES.available,
+      VEHICLE_STATES.elsewhere,
+      VEHICLE_STATES.reserved,
+      VEHICLE_STATES.on_trip,
+      VEHICLE_STATES.non_operational
+    ],
+    [VEHICLE_EVENTS.decommissioned]: [VEHICLE_STATES.removed],
+    [VEHICLE_EVENTS.provider_drop_off]: [VEHICLE_STATES.available],
+    [VEHICLE_EVENTS.unspecified]: [VEHICLE_STATES.available, VEHICLE_STATES.removed]
   }
 }
 
-const getNextState = (currStatus: VEHICLE_STATE, nextEvent: VEHICLE_EVENT): VEHICLE_STATE | undefined => {
+const getNextStates = (currStatus: VEHICLE_STATE, nextEvent: VEHICLE_EVENT): VEHICLE_STATE[] | undefined => {
   return stateTransitionDict[currStatus]?.[nextEvent]
 }
 
-const generateTransitionLabel = (
-  status: VEHICLE_STATE,
-  nextStatus: VEHICLE_STATE,
-  transitionEvent: VEHICLE_EVENT
-) => {
+// Filter for all states that have this event as a valid exiting event
+function getValidExitableStates(states: VEHICLE_STATE[], event: VEHICLE_EVENT) {
+  return states.reduce((acc: VEHICLE_STATE[], state) => {
+    if (Object.keys(stateTransitionDict[state]).includes(event)) {
+      acc.push(state)
+    }
+    return acc
+  }, [])
```

Comment:
I think this would be much more expressive, and concise by using a `.filter()` as opposed to a `.reduce()`. 
```suggestion
  return states.filter(state=> Object.keys(stateTransitionDict[state]).includes(event)))
```